packages/api-derive/test/e2e/promise.spec.ts

```diff
@@ -171,5 +167,45 @@ describe.skip('derive e2e', () => {
         done();
       });
     });
+
+    // FIXME This test doesn't pass
+    // https://github.com/polkadot-js/api/issues/1060
```

Comment:
Can we please not add it, then? Or just put it in an PR for fixing the issue mentioned in the comment?

I'm just working on making all e2e tests pass again at least in 1.0 and master.
I fixed all the derive ones in https://github.com/polkadot-js/api/pull/1084 and now I'm on the contract tests, the child storage tests and some other ones that were failing.

If we add tests, they should pass or the methods should be fixed if they're broken.

Reply:
Making that test pass is the whole point of this PR - so the comment will be removed as soon as it does.

Reply:
Ah, that wasn’t clear from the title, I thought it was primarily about logging/ error handling. Looking forward to it, then!

Reply:
The initial fix part was reverted when we realised we need more work and it is not quite the correct solution. (The test actually did pass on it, however it had some effects on API users which we are not keen on)